Sources/System/Process/TaskInfo.swift

```diff
@@ -0,0 +1,82 @@
+
+// FIXME(DO NOT MERGE): system_foo wrappers for these and mocking
+import CSystem
+
+extension ProcessID {
+  public struct TaskInfo: RawRepresentable/*, Hashable, Codable*/ {
+    /// The raw C process id.
+    @_alwaysEmitIntoClient
+    public let rawValue: CInterop.ProcTaskInfo
+
+    /// Creates a strongly-typed process id from a raw C pid
+    @_alwaysEmitIntoClient
+    public init(rawValue: CInterop.ProcTaskInfo) { self.rawValue = rawValue }
+
+    fileprivate init(_ rawValue: CInterop.ProcTaskInfo) {
+      self.init(rawValue: rawValue)
+    }
+  }
+  public func getTaskInfo() throws -> TaskInfo {
+    var result = TaskInfo.RawValue()
+    try withUnsafeMutableBytes(of: &result) {
+      let val = proc_pidinfo(self.rawValue, PROC_PIDTASKINFO, 0, $0.baseAddress, Int32($0.count))
+      // What is this wacky shenanigans?
+      guard MemoryLayout<TaskInfo.RawValue>.stride == val else {
+        throw Errno(rawValue: val)
+      }
+    }
+    return TaskInfo(result)
+  }
```

Comment:
I don't know what the intended scope is for this pull request, but a common need is to get the path and name for a PID. `proc_pidpath` and `proc_name`.